examples/fargate-serverless/main.tf

```diff
@@ -0,0 +1,232 @@
+
+  exec {
+    api_version = "client.authentication.k8s.io/v1alpha1"
+    command     = "aws"
+    # This requires the awscli to be installed locally where Terraform is executed
+    args = ["eks", "get-token", "--cluster-name", module.eks_blueprints.eks_cluster_id]
+  }
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = module.eks_blueprints.eks_cluster_endpoint
+    cluster_ca_certificate = base64decode(module.eks_blueprints.eks_cluster_certificate_authority_data)
+
+    exec {
+      api_version = "client.authentication.k8s.io/v1alpha1"
+      command     = "aws"
+      # This requires the awscli to be installed locally where Terraform is executed
+      args = ["eks", "get-token", "--cluster-name", module.eks_blueprints.eks_cluster_id]
+    }
+  }
+}
+
+data "aws_caller_identity" "current" {}
+
+locals {
+  name   = basename(path.cwd)
+  region = "us-west-2"
+
+  tags = {
+    Blueprint  = local.name
+    GithubRepo = "github.com/aws-ia/terraform-aws-eks-blueprints"
+  }
+}
+
+#---------------------------------------------------------------
+# EKS Blueprints
+#---------------------------------------------------------------
+module "eks_blueprints" {
+  source = "../.."
+
+  cluster_name    = local.name
+  cluster_version = "1.21"
+
+  vpc_id             = module.vpc.vpc_id
+  private_subnet_ids = module.vpc.private_subnets
+
+  # https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/485
+  # https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/494
+  cluster_kms_key_additional_admin_arns = [data.aws_caller_identity.current.arn]
+
+  fargate_profiles = {
+    # Providing compute for default namespace
+    default = {
+      fargate_profile_name = "default"
+      fargate_profile_namespaces = [
+        {
+          namespace = "default"
+      }]
+
+      subnet_ids = module.vpc.private_subnets
+    }
+    # Providing compute for kube-system namespace where core addons reside
+    kube_system = {
+      fargate_profile_name = "kube-system"
+      fargate_profile_namespaces = [
+        {
+          namespace = "kube-system"
+      }]
+
+      subnet_ids = module.vpc.private_subnets
+    }
+    # Add additional profiles to suit your needs
+  }
+
+  tags = local.tags
+}
+
+module "eks_blueprints_kubernetes_addons" {
+  source = "../../modules/kubernetes-addons"
+
+  eks_cluster_id = module.eks_blueprints.eks_cluster_id
+
+  enable_amazon_eks_vpc_cni = true
+  amazon_eks_vpc_cni_config = {
+    addon_version     = data.aws_eks_addon_version.latest["vpc-cni"].version
+    resolve_conflicts = "OVERWRITE"
+  }
+
+  enable_amazon_eks_kube_proxy = true
+  amazon_eks_kube_proxy_config = {
+    addon_version     = data.aws_eks_addon_version.latest["kube-proxy"].version
+    resolve_conflicts = "OVERWRITE"
+  }
+
+  enable_amazon_eks_coredns = true
+  coredns_use_managed_addon = false
+  coredns_helm_config = {
+    # Sets the correct annotations to ensure the Fargate provisioner is used and not the EC2 provisioner
```

Comment:
what's the reason for coredns not be the lastest version by default?

Reply:
It matches the API by using the default version. Safer route for users